src/menu/menu.c

```diff
@@ -761,16 +789,47 @@ menu_init(struct server *server)
 	validate(server);
 }
 
-void
-menu_finish(struct server *server)
+static void
+nullify_item_pointing_to_this_menu(struct menu *menu)
 {
+	struct menu *iter;
+	wl_list_for_each(iter, &menu->server->menus, link) {
+		struct menuitem *item;
+		wl_list_for_each(item, &iter->menuitems, link) {
+			if (item->submenu == menu) {
+				item->submenu = NULL;
+				return;
+			}
+		}
+	}
+}
```

Comment:
Can we have multiple items point to the same submenu? In that case we should likely not early return here.